ecc_enclave/enclave/crypto.h

```diff
@@ -23,3 +23,4 @@ int decrypt_state(sgx_aes_gcm_128bit_key_t* key,
     uint32_t cipher_len,
     uint8_t* plain,
     uint32_t plain_len);
```

Comment:
hmm, couldn't you just have defined that inside your own chaincode? I don't think crypto.h should be exported as part of the shim.  Eventually we want to have some extension to shim.h which provides randomness, but i think we might need a bit more discussion on API and design thereof ...

Reply:
since you did, I certainly could too!
Obviously this dirties the abstraction either above in the chaincode (your case) or below in the shim (my case).

Main reasons to keep this:
* we need it in the infrastructure (and this is the second application already that needs it). Yes, we will need discussions on the randomness thing. 
* crypto already uses it, and it is already interfaced with the SGX sdk. So, why exposing that also to the chaincode!

Reply:
> since you did, I certainly could too!
> Obviously this dirties the abstraction either above in the chaincode (your case) or below in the shim (my case).
> 
> Main reasons to keep this:
> 
> * we need it in the infrastructure (and this is the second application already that needs it). Yes, we will need discussions on the randomness thing.
> * crypto already uses it, and it is already interfaced with the SGX sdk. So, why exposing that also to the chaincode!
> 
> Actually, now I see that: (1) the crypto stuff could use it; (2) crypto does not check the return code of the rng (I'll put an issue once merged).

My main objection is that you do it in crypto.h.  The main (only) interface fpc chaincode is supposed to have is shim.h.  Obviously, in absense of a module system in C (or some build trickery to remove symbols from libraries) nothing stops CC to use symbols from other header-files/libraries right now but might not work in future when we might have a binary SDK which would (should) not contain internal header files.

Reply:
I have removed the crypto header dependency.

Reply:
Well, you haven't removed the dependency on the shim internals/crypto ;-) If you want to leave that, then maybe leaving the include as before and add a comment on why this is kind of unkosher (and what the current constraint is) would seem cleaner.  Or we just take the current get_random_bytes api and add it as-is to shim.h as first stab on such a shim API function (more or less with description that this functions is supposed to give you randomness which is identical if run on different peers iff is on exactly same CC with same state and input or otherwise the randomness is independent.

All that said probably not worth spending too much time on it now for the demo (although i think a comment would still be useful)

Reply:
alright, dependency on `shim.h`
added comment